File: lab3_2.cpp
#include <iostream>
#include <cstdlib>
#include <string.h>

using namespace std;

class Student
{
private:
    string fname;
    string lname;
    string CNP;
    int birth_year;

    string faculty_name;
    int faculty_founding_year;

public:
    void set_student_data(string fname, string lname, int birth_year, string CNP, string faculty_name, int founding_year)
    {
        this->fname = fname;
        this->lname = lname;
        this->birth_year = birth_year;
        this->CNP = CNP;
        this->faculty_name = faculty_name;
        this->faculty_founding_year = founding_year;
    }

    string get_student_fname()
    {
        return this->fname;
    }

    string get_student_lname()
    {
        return this->lname;
    }

    int get_student_birth_year()
    {
        return this->birth_year;
    }

    string get_student_CNP()
    {
        return this->CNP;
    }

    string get_faculty()
    {
        return this->faculty_name;
    }

    int get_founding_year()
    {
        return this->faculty_founding_year;
    }

    string gender_by_CNP()
    {
        char identifier = this->get_student_CNP()[0];

        if (identifier == '1' || identifier == '5')
        {
            return "male student \n";
        }

        if (identifier == '2' || identifier == '6')
        {
            return "female student \n";
        }

        else
            return "idk man";
    }

    /// deja am un getter care face asta,
    /// dar in cerinta este o metoda
    int student_age()
    {
        return (this->birth_year);
    }

    int year_gap()
    {
        return (abs((this->faculty_founding_year) - this->birth_year));
    }
};

int main()
{
    Student *MyStudent1 = new Student();
    Student *MyStudent2 = new Student();

    MyStudent1->set_student_data("Ion", "Popescu", 2000, "1930806378260", "ACS", 1967);
    MyStudent2->set_student_data("Maria", "Popescu", 2001, "2931110326224", "ACS", 1967);

    cout << MyStudent1->gender_by_CNP();
    cout << MyStudent2->gender_by_CNP();

    cout << MyStudent1->year_gap() << "\n";
    cout << MyStudent2->year_gap();

    return 0;
}

File: labs/lab3_2.kt
class Student(
    private val firstName: String,
    private val lastName: String,
    private val birthYear: Int,
    private val CNP: String,
    private val faculty: String,
    private val facultyYear: Int
) {
    fun getFirstName(): String {
        return firstName
    }

    fun getLastName(): String {
        return lastName
    }

    fun getBirthYear(): Int {
        return birthYear
    }

    fun getFaculty(): String {
        return faculty
    }

    fun getFacultyYear(): Int {
        return facultyYear
    }

    fun calculateAge(): Int {
        return (2023 - getBirthYear())
    }

    fun genderByCNP(): String {
        return if (CNP[0] == '1' || CNP[0] == '5') {
            "male"
        } else if (CNP[0] == '2' || CNP[0] == '6') {
            "female"
        } else {
            "idk"
        }
    }

    fun yearGap(): Int {
        return (getBirthYear() - getFacultyYear())
    }
}

fun main() {
    val student1 = Student("Ion", "Popescu", 2000, "1930806378260", "ACS", 1967)
    val student2 = Student("Maria", "Popescu", 2001, "6931110326224", "ACS", 1967)

    println("the student is: ${student1.genderByCNP()}, ${student1.calculateAge()} years of age, studying at ${student1.getFaculty()} ${student1.yearGap()}, years older")
    println("the student is: ${student2.genderByCNP()}, ${student2.calculateAge()} years of age, studying at ${student2.getFaculty()} ${student2.yearGap()}, years older")
}


